### PR TITLE
RPC for head info and segment of chain

### LINKED ISSUE #1772: Add RPC method to get the blockId and height of the current best block
## Description:
The current `topl_head` rpc method gives the blockId, height, score, difficulty, and entire block data. We need a convenient method that just gets the blockId and height

## Acceptance Criteria:
* Add the `topl_headInfo` method
* Add test for the new method
* Add the request to the postman collection

┆Issue is synchronized with this [Jira Task](https://topl.atlassian.net/browse/CORE-1026) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by jingsting**
## Purpose
- We need RRC methods for fetching the id and height of the best block without the other data about the block
- It way be convenient if we can retrieve a segment of the chain between a start height and an end height

## Approach
- Added a `topl_headInfo` RPC method
- Added a `topl_blocksInRange` RPC method

## Tests
- Added a test for `topl_headInfo` route
- Added a test for retrieving blocks with given start height and end height
- Added a test for the error response when start height and end height are incorrect (e.g. start height is greater than end height)
- Added a test for the response of the `topl_info` route

## Closes
#1772 
#1565

### CODE REVIEW

File: node/src/main/scala/co/topl/rpc/handlers/package.scala [hunk 1]
```diff
@@ -19,4 +19,17 @@ package object handlers {
...
     } yield keys
 
+  private[handlers] def checkHeightRange(
+    bestBlockHeight: Long,
+    startHeight:     Long,
+    endHeight:       Long
+  ): Either[RpcError, (Long, Long)] =
+    for {
+      _ <- Either.cond(
+        startHeight >= 1 && endHeight >= startHeight && bestBlockHeight >= startHeight,
```
Comment (scasplte2) [inline node/src/main/scala/co/topl/rpc/handlers/package.scala:29]: what happens when bestBlockHeight < endHeight ? Shouldn't the last check be bestBlockHeight >= endHeight?

File: topl-rpc/src/main/scala/co/topl/rpc/ToplRpcErrors.scala [hunk 2]
```diff
@@ -10,6 +10,8 @@ object ToplRpcErrors {
 
   val NoBlockIdsAtHeight: CustomError = CustomError(-32000, "No block ids found from that block height")
 
+  val InvalidHeightRange: CustomError = CustomError(-32009, "Invalid height range")
```
Comment (scasplte2) [inline topl-rpc/src/main/scala/co/topl/rpc/ToplRpcErrors.scala:13]: didn't we also want to add a check that the user is not requesting too many blocks? Like the difference between end - start <=100. I also wonder if we should also add the `blockIdsInRange` so they can get back either a list of blocks of a list of ids

**Review by scasplte2 — CHANGES_REQUESTED** [review]
please address the
- logic for getting blocks compared to best block
- error when the user specifies an end height greater than the current height
- error when the user asks for too many blocks
- add a `blockIdsInRange` method

I would say limit the number of blocks from `blocksInRange` to be 100 (configurable via settings) and the number of block ids in the `blockIdsInRange` to be 1000 (also configurable)

File: topl-rpc/src/main/scala/co/topl/rpc/ToplRpcErrors.scala (hunk 2 above)
Comment (jingsting) [reply topl-rpc/src/main/scala/co/topl/rpc/ToplRpcErrors.scala:13]: That was added in the methodAlias-rpc branch, sorry it got mixed into that one

Comment (scasplte2) [reply topl-rpc/src/main/scala/co/topl/rpc/ToplRpcErrors.scala:13]: I just saw that. Thanks for letting me know 👍

File: node/src/main/scala/co/topl/rpc/handlers/package.scala (hunk 1 above)
Comment (jingsting) [reply node/src/main/scala/co/topl/rpc/handlers/package.scala:29]: We will get back blocks from startHeight to bestBlockHeight if bestBlockHeight < endHeight. Do you think we need to enforce getting the exact range of blocks? I could add that condition here

**Comment by jingsting** [comment]
> please address the
> 
> * logic for getting blocks compared to best block
> * error when the user specifies an end height greater than the current height
> * error when the user asks for too many blocks
> * add a `blockIdsInRange` method
> 
> I would say limit the number of blocks from `blocksInRange` to be 100 (configurable via settings) and the number of block ids in the `blockIdsInRange` to be 1000 (also configurable)

- added a check enforcing that bestBlockHeight >= endHeight
- the checks for block/tx number limit were added in a different branch
- `topl_blockIdsInRange` was added in `blockIdsInRange-rpc` branch

**Review by scasplte2 — APPROVED** [review]
🔥

**Pull request merged**